mlflow/models/evaluation/utils/trace.py

```diff
@@ -0,0 +1,92 @@
+import contextlib
+import inspect
+import logging
+
+import mlflow
+from mlflow.ml_package_versions import FLAVOR_TO_MODULE_NAME
+from mlflow.utils.autologging_utils import (
+    AUTOLOGGING_CONF_KEY_IS_GLOBALLY_CONFIGURED,
+    AUTOLOGGING_INTEGRATIONS,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def configure_autologging_for_evaluation(enable_tracing: bool = True):
```

Comment:
Does it take long to run all `autolog`?

Reply:
Depends on which ML libraries are installed. It can be a bit long when **all** ML libraries are installed, which was about 8 seconds in my test environment.

```
start = time.time()
with configure_autologging_for_evaluation():
    pass
end = time.time()
print(end - start)

# > 8.47714900970459
```

However, I think it is extremely rare that user's environment has all flavor libraries installed. For example, in another test env where `scikit-learn`, `langchain`, and `openai` are installed, it took 1 second, which is be ignorable latency compared to running evaluation.

Reply:
Does it take 8 seconds on Databricks?

Reply:
In OSS. I've tested in MLR and it took 4 seconds, as we actually many libraries pre-installed (1) openai (2) transformers (3) torch (4) langchain (5) xgboost (6) scikit-learn (7) sentence-transformers (8) lightgbm

Reply:
Does it always 4 seconds? What happens if we run evaluate multiple times?

Reply:
No, it takes 4 seconds for the first time only. Second, third calls are much quicker (less than a second).

Reply:
@harupy Do you think this is a blocker? I'm not sure how we can reduce the time as it mainly comes from library import tho.

Reply:
Discussed offline, we merge this for unblocking release but optimize this before the RC release (ideally to sub-second latency). Here are possible approaches in my mind:
* For libraries that are not imported already, use post-import hook (like `mlflow.autolog()`) instead of enable/disable it immediately.
* Limit the scope to trace-enabled libraries. For other flavors, use the global `AUTOLOGGING_GLOBALLY_DISABLED` kill switch instead, which was the original method used before this PR.